nx/lib/nx/binary_backend.ex

```diff
@@ -2153,6 +2153,72 @@ defmodule Nx.BinaryBackend do
     from_binary(output, new_data)
   end
 
+  @impl true
+  def fft(out, %{shape: {n}} = tensor) do
+    data =
+      match_types [tensor.type] do
+        for <<match!(x, 0) <- to_binary(tensor)>> do
+          read!(x, 0)
+        end
+      end
+
+    result =
+      if rem(n, 2) == 0 do
+        fft_list(data)
+      else
+        dft_list(data)
+      end
+
+    output_data =
+      match_types [out.type] do
+        for item <- result, into: <<>> do
+          <<write!(item, 0)>>
+        end
+      end
+
+    from_binary(out, output_data)
+  end
+
+  defp dft_list(data) do
+    bigN = length(data)
+
+    for k <- 0..(bigN - 1) do
+      {xk, _} =
+        Enum.reduce(data, {0, 0}, fn x, {acc, n} ->
+          {acc + x * Complex.exp(Complex.new(0, -2 * :math.pi() * k * n / bigN)), n + 1}
+        end)
+
+      xk
+    end
+  end
+
+  defp fft_list([]), do: []
+  defp fft_list([data]), do: [data]
```

Comment:
We should have this clause for `dft_list`, for fft we don't expect an odd number :)

Reply:
This is the base case for recursion

Reply:
Oh, I think we should be calling the top-most fft recursively, rather than `fft_list` directly. For example if the length is 6, it splits into 3 and 3, and for those we need to use `dft_list`.

We could actually make it a single function with guards.

Reply:
I chose to split because I think it's a little messy to use different datatypes and signatures for the same function name ina recursion like this. So the top level formats the data and then calls this.

But I agree we can use a guard to merge dft_list into fft_list!

Reply:
The datatype and signature for `dft_list` and `fft_list` is the same though, it's just that `fft_list` is a special case as an optimisation, no? My point main point is that currently it doesn't work for 6 (I think) :)